### ISSUE TITLE
$.post is not sending the data

### ISSUE THREAD

**Issue by elpepefontana** [olton/Metro-UI-CSS#1781]
I have created the next function:
`

function ajaxPostCall(objectName, method, send) {
    return $.post(setURL(), {object: objectName, method: method, dataToSend: send});
}

function setURL() {
    return homePath + '/AjaxHandler/';
}

`
I have a page that use a metro 4 modified table, image provided:

<img width="1653" alt="Captura de Pantalla 2021-05-24 a la(s) 16 31 07" src="https://user-images.githubusercontent.com/2715358/119398368-d68f1f00-bcad-11eb-9e4d-8d5031a0e79d.png">

This page calls for the table content using the function ajaxPostCall, which returns a promise.
Then the same function is used to send the request that is used to draw the form.
The first time all goes ok, and the table gets drawn ok.
Later when i call the function to draw the form, no DATA is sent on the post.
I see no difference between one call or another. The params sent by each, are ok.

I'm using:
 - OS: mac
 - Browser Chrome
 - Metro Version 4.4.3

The next function is the one that produce the error, when calling the ajaxPostCall function:
`

function drawForm(action, objectName, context, dataToSend) {
    var father = readHiddenData(objectName + '_father');
    var parent = readHiddenData(objectName + '_id');
    
    dependData = {}; // keep it as global - mantener como global
    
    var send = {};
    send.actionType = action;
    send.formName = objectName;
    send.formFather = !isEmpty(father.name) ? father.name : null;
    send.formMethod = 'POST';
    send.formAction = '';
    send.formThreshold = 10;
    send.objectName = objectName;
    send.formData = dataToSend;
    send.parentId = !isEmpty(parent.id) ? parent.id : null;
    
    var ifJsonFormDataCreated = ajaxPostCall(objectName, 'createJsonForm', send);
    
    ifJsonFormDataCreated.then(
        function(response) {
            var formData = JSON.parse(response);
            createForm(formData, context, action);
            createHiddenData(objectName + '_fields', 'hiddenData', formData.fields);
        },
        function(xhr){
            console.log(xhr.status, xhr.statusText, xhr.responseText);
        }    
    );
}

`
<img width="1525" alt="Captura de Pantalla 2021-05-24 a la(s) 16 49 09" src="https://user-images.githubusercontent.com/2715358/119405853-4dc9b080-bcb8-11eb-810a-78866779513e.png">

The data appears to be sent but no data passes trough the post send.
On php side no data is received,  i get an error for not finding the 'object' index on the data sent.

Im adding images of my Fiddler everywhere.

I've tried to replace the ajaxPostCall with the direct call to $.post, but the error remains.
Later, I've tried to use the $.ajax call using events, from that i get the error on the object index of the sent data.

I have no clue of what is happening...

**Comment by elpepefontana** [comment]
Hello, again.
I've eliminated the function ajaxPostCall, and changed all the functions that used it to use a direct call to $.post(), but having the same problem, i still, don't get the data on the server side.

I've noticed that if i print the values of the data sent to the browser, the first time the script calls the $.post call, the table is not rendered, due to a bad formed json response, when i click the 'green' (the "new register" button) button the values are sent to the server side of the script.

In your experience haw can, anothers parts of the code affect the function call??? all the parameters to send are ok, when i click the button.
